README.md

```diff
@@ -1 +1,167 @@
+
+Let's say you have to implement three HTTP requests to a third-party service (like Facebook or Google), but they need to be performed in a sequential order, because the second and third request use information from the response of the first request (e.g., API token).  We can't delay code the way we do in traditional synchronous style, so you end up with nested callbacks to ensure the right order of execution. This is the pseudo code:
+
+```js
+callOne({...}, function(error, data1) {
+    callTwo(data1, function(error, data2) {
+        callThree(data2, function(error, data3) {
+        ...
+        })
+    })
+})
+```
+
+Remember, this is asynchronous world so we cannot type `callOne` then `callTwo` on a new line. We need to use callback and with callbacks (the way they are written in the example above) comes indentation.
+
+Now add on top of that a few callbacks to perform something else essential like opening a database connection and starting a server. Look at this:
+
+
+```js
+fs.readdir(source, function (err, files) {
+  if (err) {
+    console.log('Error finding files: ' + err)
+  } else {
+    files.forEach(function (filename, fileIndex) {
+      console.log(filename)
+      gm(source + filename).size(function (err, values) {
+        if (err) {
+          console.log('Error identifying file size: ' + err)
+        } else {
+          console.log(filename + ' : ' + values)
+          aspect = (values.width / values.height)
+          widths.forEach(function (width, widthIndex) {
+            height = Math.round(width / aspect)
+            console.log('resizing ' + filename + 'to ' + height + 'x' + height)
+            this.resize(width, height).write(dest + 'w' + width + '_' + filename, function(err) {
+              if (err) console.log('Error writing file: ' + err)
+            })
+          }.bind(this))
+        }
+      })
+    })
+  }
+}
+```
+
+You might end up with something called the nested approach, pyramid of doom or callback hell. This code is very hard to read and maintain. People who have to work on this code after you might start to hate you!
+
+The `readdir` example is from <http://callbackhell.com> (yes, there is even a website for Callback Hell!).
+
+This lesson will cover the ways to combat callback hell.
+
+## Objectives
+
+1. Explain that a callback hell script is a complication of the callback async pattern
+1. Identify a callback hell script
+1. Implement possible ways to combat callback hell with named functions
+1. Implement possible ways to combat callback hell with modules
+
+
+## Using A Named Function to Mitigate Callback Hell
+
+The easiest way to reduce the level of indentation and thus reduce callback hell is to use named functions. Named functions are functions with names or variables which values are functions.
+
+```js
+function processResponse1(error, data1) {
```

Comment:
I agree with @vicfriedman that it would be helpful for these examples to be fully fleshed out and that the students should be able to code along. Could we pull in an example from, say, a Google API OAuth flow? Refreshing auth tokens and then taking an action can be a major pain if done entirely with callbacks, and it will really help students to understand how to approach structuring a Node.js workflow.